modules/images/src/lib/parsers/parse-to-image-bitmap.js

```diff
@@ -4,9 +4,33 @@
 // Supported on worker threads, not supported on Edge, IE11 and Safari
 // https://developer.mozilla.org/en-US/docs/Web/API/ImageBitmap#Browser_compatibility
 
+let imagebitmapOptionsSupported = true;
+
 export default function parseToImageBitmap(arrayBuffer, options) {
   // NOTE: In some cases unnecessary conversion to blob (response, blob, file input)
   const blob = new Blob([new Uint8Array(arrayBuffer)]); // MIME type not needed...
-  const imagebitmapOptions = options && options.imagebitmap;
-  return imagebitmapOptions ? createImageBitmap(blob) : createImageBitmap(blob, imagebitmapOptions);
+  let imagebitmapOptions = options && options.imagebitmap;
+
+  // Firefox crashes if imagebitmapOptions is supplied
+  // Avoid supplying if not provided, track if not supported
+  if (isEmptyObject(imagebitmapOptions) || !imagebitmapOptionsSupported) {
+    imagebitmapOptions = null;
+  }
+
+  if (imagebitmapOptions) {
+    try {
+      return createImageBitmap(blob, imagebitmapOptions);
+    } catch (error) {
+      console.warn(error); // eslint-disable-line
+      imagebitmapOptionsSupported = false;
+    }
+  }
+  return createImageBitmap(blob);
+}
+
+function isEmptyObject(object) {
+  for (const key in object) {
```

Comment:
This doesn’t seem to account for object===undefined

Reply:
> This doesn’t seem to account for object===undefined

Shouldn't happen due to the way options are merged with defaults that said adding the check doesn't cost much., will do.

Reply:
Looking at the previous attempt: https://github.com/uber-web/loaders.gl/commit/d348845a489739a1464e5e27ac6c5e8764cdd041

Does it throw immediately or does the promise reject?

Reply:
> Does it throw immediately or does the promise reject?

Yes I believe you have a point. `async await` converts automatically between promise rejections and exceptions, so I'll update to use async syntax.

Still this makes it obvious that I have not been able to test these changes, due to our test harness limitations.

Reply:
Could you hold on merging while I fix the testing?